openg2p_registration/models/openg2p_create_registration.py

```diff
@@ -0,0 +1,319 @@
+                if temp["autodedup_id"] == "1":
+                    # merging the existing ones with new one and passing the existing_id and current_id as arguments
+                    self.merge_registrations(temp, kycdup_list[0].id)
+                elif temp["autodedup_id"] == "2":
+                    # deleting the old record
+                    duplicate_kyc = self.env["openg2p.registration"].search(
+                        [("kyc_id", "=", temp["bban"])]
+                    )
+                    duplicate_kyc.active = False
+                    duplicate_kyc_bene = self.env["openg2p.beneficiary"].search(
+                        [("kyc_id", "=", temp["bban"])]
+                    )
+                    duplicate_kyc_bene.active = False
+                    self.refactor_create_fields(temp)
+                elif temp["autodedup_id"] == "3":
+                    print("kyc and delete new")
+                    # deleting the current record
+            else:
+                self.refactor_create_fields(temp)
+        elif int(temp["autodedup_id"]) in range(4, 7):
+            externaldup_list = (
+                self.env["openg2p.registration"]
+                .search([("external_id", "=", temp["new_emis_code"])])
+                .ids
+            )
+            if len(externaldup_list) != 0:
+                if temp["autodedup_id"] == "4":
+                    # merging the existing ones with new one and passing the existing_id and current_id as arguments
+                    self.merge_registrations(temp, externaldup_list[0])
+                elif temp["autodedup_id"] == "5":
+                    duplicate_external = self.env["openg2p.registration"].search(
+                        [("external_id", "=", temp["new_emis_code"])]
+                    )
+                    duplicate_external.active = False
+                    duplicate_external_bene = self.env["openg2p.beneficiary"].search(
+                        [("external_id", "=", temp["new_emis_code"])]
+                    )
+                    duplicate_external_bene.active = False
+                    self.refactor_create_fields(temp)
+                elif temp["autodedup_id"] == "6":
+                    print("External id & delete new")
+            else:
+                self.refactor_create_fields(temp)
+        else:
+            self.refactor_create_fields(temp)
+
+    def refactor_create_fields(self, temp):
+        country_name = temp["country"] if "country" in temp.keys() else "Sierra Leone"
+        state_name = temp["state"] if "state" in temp.keys() else "Freetown"
+        country_id = self.env["res.country"].search([("name", "=", country_name)])[0].id
+        state_id = (
+            self.env["res.country.state"].search([("name", "=", state_name)])[0].id
+        )
+        # Creating registration in final stage
+        stage_id = 6
+        temp["stage_id"] = int(temp["next_stage"])
+        try:
+            regd = self.create(
```

Comment:
Please explain why this new piece of creating a registration is added with such values.